nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/is_new_path_shorter.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 Pradheep Padmanabhan - Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__IS_NEW_PATH_SHORTER_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__IS_NEW_PATH_SHORTER_HPP_
+
+#include <string>
+#include <memory>
+#include <limits>
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "behaviortree_cpp_v3/decorators/retry_node.h"
+
+namespace nav2_behavior_tree
+{
+
+/**
+ * @brief A BT::DecoratorNode that ticks its child everytime when the length of
+ * the new path is smaller than the old one by the length given by the user.
+ */
+class IsNewPathShorter : public BT::RetryNode
+{
+public:
+  /**
+   * @brief A constructor for nav2_behavior_tree::IsNewPathShorter
+   * @param name Name for the XML tag for this node
+   * @param conf BT node configuration
+   */
+  IsNewPathShorter(
```

Comment:
I think this needs a slightly more descriptive name which indicates the goal distance element of this as well.